src/api/torii-endpoints.md

```diff
@@ -152,6 +152,14 @@ Learn [how to use metrics](/guide/advanced/metrics).
 | Permission Error | 403    | [`QueryExecutionFail::Permission(String)`](/api/data-model-schema#queryexecutionfail)      |
 | Find Error       | 404    | [`QueryExecutionFail::Find(FindError)`](/api/data-model-schema#queryexecutionfail)         |
 
+
+::: tip Lazily Evaluated Pagination
+
+TODO: explain how it works. Explain that this behaviour is configured with [`torii.query_results_per_fetch`](/api/config/torii-params#torii-query-results-per-fetch) and [`torii.query_idle_time`](/api/config/torii-params#torii-query-idle-time).
```

Comment:
@mversic please help

Reply:
@arjentix has been making changes recently. I think he can give more correct overview

Reply:
Hey, sorry for the delay.

So basically after https://github.com/hyperledger/iroha/pull/3958 users should send either `SignedQuery` with `Pagination` and `Sorting` or just `ForwardCursor`.

`SignedQuery` -- an Iroha query signed by a user keypair
`Pagination` -- struct with two numbers: limit of indexing and start of indexing. If both are zeros, then pagination is not applied
`Sorting` -- currently we support sorting only by a metadata key of an asset metadata. So that user can place `Name` (string with some contrains on its content) inside and then all output assets will be sorted by a value of the specified key if there is some. `Sorting` can contain nothing as well, in that case it won't be applied.
`ForwardCursor` -- at first you don't have it. But if the output of your query is a big array, then you'll get a cursor from Iroha with a **part** of the answer. To get the other parts you should provide this cursor to Iroha in a new request. Repeat this step until cursor is depleted (you'll know that by getting empty cursor or error).

I think you can use this info to provide a good documentation.

Reply:
Moved: https://github.com/hyperledger/iroha-2-docs/issues/426